services/dedup/scylla/scylla.go

```diff
@@ -33,6 +33,63 @@ func (d *ScyllaDB) Close() {
 	d.scylla.Close()
 }
 
+func (d *ScyllaDB) GetBatch(kvs []types.KeyValue) (map[types.KeyValue]bool, map[types.KeyValue]int64, error) {
+	defer d.stat.NewTaggedStat("dedup_get_batch_duration_seconds", stats.TimerType, stats.Tags{"mode": "scylla"}).RecordDuration()()
+	// Prepare a map to store results for each job (true = accept, false = reject)
+	results := make(map[types.KeyValue]bool)
+	sizes := make(map[types.KeyValue]int64)
+
+	// Group jobs by workspaceID for batch querying
+	workspaceJobsMap := make(map[string][]types.KeyValue)
+	d.cacheMu.Lock()
+
+	for _, kv := range kvs {
+		if previous, found := d.cache[kv.Key]; found {
+			results[kv] = false
+			sizes[kv] = previous.Value
+			continue
+		}
+		d.cache[kv.Key] = kv
+		results[kv] = true
+		sizes[kv] = kv.Value
+		workspaceJobsMap[kv.WorkspaceID] = append(workspaceJobsMap[kv.WorkspaceID], kv)
+	}
+	d.cacheMu.Unlock()
+
+	// Loop over each workspace and query all messageIDs for that workspace
+	for workspaceID, workspaceJobs := range workspaceJobsMap {
+		var messageIDs []string
+		for _, job := range workspaceJobs {
+			messageIDs = append(messageIDs, job.Key)
+		}
+
+		messageIDChunks := lo.Chunk(messageIDs, d.batchSize)
```

Comment:
we should keep the read batch size and write batch size separate

Reply:
Done